sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/MapperSpec.kt

```diff
@@ -128,6 +119,46 @@ internal class MapperSpec private constructor(private val nameAllocator: NameAll
     return factoryFieldname
   }
 
+  private fun QueryResults.cursorGetter(
+      mapper: TypeSpec.Builder,
+      constructor: MethodSpec.Builder,
+      typePrefix: String = ""
+  ): CodeBlock {
+    val creatorField = if (!isView) Table.CREATOR_FIELD else "$queryName${Table.CREATOR_CLASS_NAME}"
+
+    // For foreign tables we need to add their factory as a field so it can be used
+    // during mapping. Note that values can also have foreign tables if they are
+    // custom types grabbed from a foreign table.
+    foreignTypes().forEachIndexed { i, foreignTable ->
+      addFactoryField(mapper, constructor, "${typePrefix}R${i+1}", foreignTable)
+    }
+
+    if (isView && factoryFields.add(creatorField)) {
+      // Add the parameters and fields to the mapper/constructor
+      val type = ParameterizedTypeName.get(creatorType, TypeVariableName.get(typePrefix))
+      mapper.addTypeVariable(TypeVariableName.get(typePrefix, interfaceType))
+          .addField(type, creatorField, Modifier.PRIVATE, Modifier.FINAL)
+
+      constructor.addParameter(type, creatorField)
+          .addStatement("this.$creatorField = $creatorField")
+    }
+
+    val mapReturn = CodeBlock.builder().add("$creatorField.create(\n")
+    mapReturn.indent().indent()
```

Comment:
so i originally tried this but it didn't work when the following line also has a `$[` javapoet threw some erorr.